### DISCUSSION TITLE
RESEARCH: Channel Rebalancing Script

### DISCUSSION THREAD

**Discussion by rootzoll** [raspiblitz/raspiblitz#3913] [Research]
Take a look at this tweet:
https://twitter.com/ockeghem7/status/1102358521100808194 based on the works of:
https://github.com/C-Otto/rebalance-lnd

It would be an awesome idea to add such rebalancing scripts as a service you can switch on/off set with some parameters on the RaspiBlitz.

This thread is for everybody wants to experiment with that, share experience and maybe even implement it as a service.

For switching on/off think about a script like `/home/admin/config.scripts/lnd.rebalance.sh` - take a look how the other config scripts in that directory are written (feel free to choose python, if you dont familiar with shellscripts). Also the RaspiBlitz has a background-script `/home/admin/_background.sh` that can (when service is signaled on in the `/mnt/hdd/raspiblitz.conf`) check if a running of that rebalancing script is necessary and execute it.

**Comment by bitsam** [comment]
@rootzoll . The twitter account is now disabled. I was just reading today about the C-Otto rebalance script and would like to give it a try.

**Comment by rootzoll** [comment]
@bitsam maybe also take a look at: https://github.com/bitromortac/lndmanage

**Comment by scarolan** [comment]
This looks awesome. I would love to have something like this auto-rebalance channels for me.

**Comment by rootzoll** [comment]
Re-adding lndmanage is on our roadmap for v1.8: see #2314

**Comment by openoms** [comment]
> This looks awesome. I would love to have something like this auto-rebalance channels for me.

`bos rebalance` is working very well
Install `Balance of Satoshis` from SERVICES for LND.
C-lightning has `rebalance` plugin and also CLBOSS achieves balanced channels.

**Comment by scarolan** [comment]
@openoms do you find this relatively maintenance free?  I would like to re-activate my lightning node but being short on time I really can't afford to spend hours micromanaging my channels.
